treescript/src/treescript/merges.py

```diff
@@ -0,0 +1,180 @@
+"""Treescript merge day functionality."""
+import logging
+import os
+import shutil
+
+from scriptworker_client.utils import makedirs
+from treescript.exceptions import TaskVerificationError
+from treescript.mercurial import get_revision, run_hg_command
+from treescript.merge_config import merge_configs
+from treescript.task import get_merge_flavor
+from treescript.versionmanip import do_bump_version, get_version
+
+log = logging.getLogger(__name__)
+
+
+def replace(file_name, from_, to_):
```

Comment:
> This looks good, thanks! I've left a bunch of comments, some minor and some slight longer. The comments were written over an extended period of time, so might not all exactly hang together. I'm happy to chat about any thing here, if it is unclear.

Do you have any comments about the mercurial instructions?

> It would be in mozilla-central, not in the hook.

I couldn't reply to this comment in-line and can't see what it's a reply to, but I'm guessing it's about the version control of the configuration. I'm guessing this is because we'd be using an action task to then create the actual merge task (Grumbles about overcomplication saved for later). I'm ok as long as the configuration is under vcs somewhere, although I worry that it'll be harder to find.